### ISSUE TITLE
Significant issue in find_in_batches (perhaps next_batch_from and everything that calls it)

### ISSUE THREAD

**Issue by devnut** [cequel/cequel#370]
If there is a table with a schema with 4 keys like this:

```
    key :app_id, :text
    key :client_id, :text
    key :filter_type_id, :text
    key :filter_id, :text

```
And then you do WhateverModel::find_in_batches()

the queries end up looking something like this:

> [2017-03-21 00:08:32.877 (30545|5779200) D]  CQL (78ms) SELECT * FROM whatever_models LIMIT 1000
> [2017-03-21 00:08:32.933 (30545|5779200) D]  CQL (24ms) SELECT * FROM whatever_models WHERE app_id = 'app_id' AND client_id > 'some_client_id' LIMIT 1000
> [2017-03-21 00:08:32.991 (30545|5779200) D]  CQL (2ms) SELECT * FROM whatever_models WHERE TOKEN(app_id) > TOKEN('app_id') LIMIT 1000
> 

Note that it is only looking at app_id and client_id.  It is completely ignoring filter_type_id and filter_id when constructing the queries to range through the table.

I'm pretty sure this is happening in next_batch_from, but I can't quite figure out whats going on in there yet

**Comment by pezra** [comment]
Yeah, that code is pretty complicated. It was necessary in the distant past
but these days we should use the paged results capabilities of c*. I'd
accept a PR to that effect if you are interested in making such a change.

**Comment by kuldeepaggarwal** [comment]
@pezra Can you give some tip from where to start, I want to contribute but still a newbie?

**Comment by pezra** [comment]
The problem is somewhere in [#next_batch_from](https://github.com/cequel/cequel/blob/master/lib/cequel/record/record_set.rb#L722). That function should use [result paging](http://docs.datastax.com/en/developer/ruby-driver/3.2/features/basics/result_paging/) instead of building a new query.
